index.js

```diff
@@ -22,7 +27,16 @@ module.exports = {
 
   init: function() {
     var bowerDeps = this.project.bowerDependencies();
+
     var VersionChecker = require('ember-cli-version-checker');
+    var options = this.options = this.options || {};
+
+    options.babel = options.babel || {};
+
+    if (process.env.EMBER_ENV === 'production') {
```

Comment:
Yes, it does get set automatically, but not before `init` of addons.  It is set here: https://github.com/ember-cli/ember-cli/blob/3073a5c6a67debc6f4a1c37dc404ccab126d7309/lib/models/builder.js#L45 during the starting of the build.